### PR TITLE
MESH-1255/Transaction Payment: Pull in some substrate-rc6 changes

### PR DESCRIPTION
**Pull request by Centril**
Make sure to review this with whitespace changes hidden.

### CODE REVIEW

File: pallets/transaction-payment/src/lib.rs [hunk 1]
```diff
@@ -60,12 +59,12 @@ use sp_runtime::{
...
         ValidTransaction,
     },
-    FixedI128, FixedPointNumber, FixedPointOperand, Perquintill,
+    FixedPointNumber, FixedPointOperand, FixedU128, Perquintill,
 };
 use sp_std::prelude::*;
 
 /// Fee multiplier.
-pub type Multiplier = FixedI128;
+pub type Multiplier = FixedU128;
```
Comment (Centril) [inline pallets/transaction-payment/src/lib.rs:67]: This is used in the only storage item; should we write a migration?

File: pallets/transaction-payment/src/lib.rs [hunk 2]
```diff
@@ -392,21 +464,23 @@ where
         info: &DispatchInfoOf<T::Call>,
         len: usize,
     ) -> Result<(BalanceOf<T>, Option<NegativeImbalanceOf<T>>), TransactionValidityError> {
-        let fee = Module::<T>::compute_fee(len as u32, info, 0u32.into());
+        let tip = self.0;
```
Comment (Centril) [inline pallets/transaction-payment/src/lib.rs:467]: @satyamakgec So AIUI, we want to revert all the changes in this function?

Comment (satyamakgec) [reply pallets/transaction-payment/src/lib.rs:467]: Yes in this function and some code comments which have mention of a tip. and could you also please add the tip change in the top of the pallet.

File: pallets/transaction-payment/src/lib.rs [hunk 3]
```diff
@@ -234,8 +301,8 @@ where
 {
     /// Query the data that we know about the fee of a given `call`.
     ///
-    /// As this module is not and cannot be aware of the internals of a signed extension, It only
-    /// interprets the extrinsic as some encoded value and accounts for its weight
+    /// This module is not and cannot be aware of the internals of a signed extension, for example
+    /// a tip. It only interprets the extrinsic as some encoded value and accounts for its weight
```
Comment (satyamakgec) [inline pallets/transaction-payment/src/lib.rs:305]: We remove the tip wording intentionally as we are not supporting the tip functionality in our chain.

File: pallets/transaction-payment/src/lib.rs [hunk 4]
```diff
@@ -509,7 +584,10 @@ where
                         // is gone in that case.
                         Err(_) => payed,
                     };
-                T::OnTransactionPayment::on_unbalanced(actual_payment);
+                let imbalances = actual_payment.split(tip);
+                T::OnTransactionPayment::on_unbalanceds(
+                    Some(imbalances.0).into_iter().chain(Some(imbalances.1)),
+                );
```
Comment (satyamakgec) [inline pallets/transaction-payment/src/lib.rs:590]: we also need the revert the changes here as well.

**Review by satyamakgec — CHANGES_REQUESTED** [review]

File: pallets/transaction-payment/src/lib.rs (hunk 1 above)
Comment (maxsam4) [reply pallets/transaction-payment/src/lib.rs:68]: Probably best to do that. Just to be safe. ps We don't actually use the fee multiplier. It should be set as a constant 1.

Comment (Centril) [reply pallets/transaction-payment/src/lib.rs:67]: We decided it was not necessary on a meeting.

File: pallets/transaction-payment/src/lib.rs [hunk 5]
```diff
@@ -392,21 +464,23 @@ where
...
-                &payer_key,
-                fee,
-                WithdrawReason::TransactionPayment.into(),
-                ExistenceRequirement::KeepAlive,
-            )
-            .map_err(|_| InvalidTransaction::Payment)?;
+            let reason = if tip.is_zero() {
+                WithdrawReason::TransactionPayment.into()
+            } else {
+                WithdrawReason::TransactionPayment | WithdrawReason::Tip
```
Comment (maxsam4) [inline pallets/transaction-payment/src/lib.rs]: Since tip is always 0, withdraw reason is always `WithdrawReason::TransactionPayment.into()`

Comment (Centril) [reply pallets/transaction-payment/src/lib.rs]: Yep, already fixed :)

**Review by satyamakgec — APPROVED** [review]

**Review by maxsam4 — APPROVED** [review]
